### DISCUSSION TITLE
writing a javascript library to work well with solara

### DISCUSSION THREAD

**Discussion by cleemesser** [widgetti/solara#778] [General]
Hello, I am really impressed by solara. Having written other widgets to run inside of jupyter notebooks, it really seems to address some the issues with complexity that comes up. 

I have a pretty open-ended question:

I am writing a javascript library/component that draws to html5 canvas -- a variant on a plotting widget. I do not write a lot of javascript so always have to go through relearning the language and ecosystem every few years. 

Is there any advice on things to include in my library which will make it easy to integrate with solara? What you the developers advise if you are starting from scratch? Would you organize it using react or lit or some similar JS component system/framework?

Addendum: looking through the code and dependencies, it is clear you have used vue.js a lot. But also wrap a other code bases.

**Comment by maartenbreddels** [comment]
This is a good question, and I'm working on expanding the documentation on that topic. Do you want to be a proof reader, and see if that matches what you need to know?

**Reply by cleemesser** [reply]
I would be glad to help proof read.
